### PR TITLE
docs(sdk): add utilities doc

### LINKED ISSUE #72: Maintain an api-doc for sdk utilities
Users are unaware of the utilities provided by the spore-sdk. 
To address this, we need to maintain an API documentation for the utilities.

### PR DESCRIPTION
**Pull request by ShookLyngs**
## Changes
- Added TOC to the compsed-apis doc
- Created a utilities doc to describe handy helper functions

## Related issues
- resolves #72

### CODE REVIEW

File: docs/core/utilities.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+const result2 = isContentTypeValid('image/;test=1');
+console.log(result2); // false
+```
+
+## Transaction
+
+### waitForTransaction
+
+Wait for a transaction to be committed on-chain.
+Will throw an error if the transaction is not committed within some times.
```
Comment (anipenguin) [inline docs/core/utilities.md]: How long would it take before it throws an error?

File: docs/core/utilities.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+Returns a boolean indicating weather if the target ScriptId exists in the SporeConfig.
+If "scriptName" is passed, it also checks whether the name of the target script matches.
+
+```typescript
+declare function isSporeScriptSupported(config: SporeConfig, scriptId: ScriptId, scriptName?: string): boolean;
+```
+
+#### Example
+
+Check if the target cell is any script supported by the SporeConfig:
```
Comment (anipenguin) [inline docs/core/utilities.md]: edit, grammar: "Check if the target cell is a script supported by the SporeConfig:"

File: docs/core/utilities.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+```typescript
+const sporeScript = getSporeScript(config, 'Spore', {
+  codehash: '0xbbad126377d45f90a8ee120da988a2d7332c78ba8fd679aab478a19d6c133494',
+  hashType: 'data1',
+});
+```
+
+### isSporeScriptSupported
+
+Returns a boolean indicating weather if the target ScriptId exists in the SporeConfig.
```
Comment (anipenguin) [inline docs/core/utilities.md]: edit, typo & grammar: "Returns a boolean indicating whether the target ScriptId exists in the SporeConfig."

**Review by anipenguin — CHANGES_REQUESTED** [review]

File: docs/core/utilities.md (hunk 1 above)
Comment (ShookLyngs) [reply docs/core/utilities.md]: The wait time is dynamic for each turn, I'll say approximately 2 mins.

**Review by ashuralyk — APPROVED** [review]

**Review by anipenguin — APPROVED** [review]

**Pull request merged**